### PR TITLE
fix xref with query string is not resolved.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @superyyrrzz @qinezh

### CODE REVIEW

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/HostService.cs [hunk 1]
```diff
@@ -348,10 +351,13 @@ private MarkupResult ParseCore(MarkupResult markupResult, FileAndType ft)
                                      where string.Equals(n.Name, "xref", StringComparison.OrdinalIgnoreCase)
                                      from attr in n.Attributes
                                      where string.Equals(attr.Name, "href", StringComparison.OrdinalIgnoreCase) || string.Equals(attr.Name, "uid", StringComparison.OrdinalIgnoreCase)
-                                     where !string.IsNullOrWhiteSpace(attr.Value)
+                                     let target = attr.Value
+                                     where !string.IsNullOrWhiteSpace(target)
+                                     let queryIndex = target.IndexOf("?")
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/HostService.cs]: >"?" [](start = 69, length = 3)

also handle "#"?

**Comment by hellosnow** [comment]
:shipit:

**Comment by qinezh** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Pull request merged**